packages/makeup-modal/src/index.js

```diff
@@ -3,12 +3,67 @@
 const keyboardTrap = require('makeup-keyboard-trap');
 const screenreaderTrap = require('makeup-screenreader-trap');
 
+let hoistEl;
+let hoistedElementPlaceholder;
+let containerDiv;
+let bodyChildIndexes = [];
+
+function unhoist() {
+    if (hoistEl) {
+        if (containerDiv) {
+            [...containerDiv.childNodes].forEach((child) => {
+                if (!child.src) {
+                    const index = bodyChildIndexes.shift();
+                    if (index > document.body.childNodes.length) {
+                        document.body.appendChild(child);
+                    } else {
+                        document.body.insertBefore(child, document.body.childNodes[index + 1]);
+                    }
+                }
+            });
+            containerDiv.remove();
+            containerDiv = null;
+            bodyChildIndexes = [];
+        }
+
+        if (hoistedElementPlaceholder) {
+            hoistedElementPlaceholder.replaceWith(hoistEl);
+            hoistedElementPlaceholder = null;
+        }
+        hoistEl = null;
+    }
+    return hoistEl;
+}
+
+function hoist(el) {
+    unhoist();
+    hoistEl = el;
+
+    hoistedElementPlaceholder = document.createElement('div');
+    hoistEl.parentElement.insertBefore(hoistedElementPlaceholder, hoistEl);
+
+    containerDiv = document.createElement('div');
+    [...document.body.childNodes].forEach((child, index) => {
+        if (!child.src) {
+            containerDiv.appendChild(child);
+            bodyChildIndexes.push(index);
+        }
+    });
+
+    document.body.prepend(containerDiv);
+
+    document.body.appendChild(hoistEl);
+
+    return hoistEl;
+}
+
 let modalEl;
 
 function unmodal() {
     if (modalEl) {
-        screenreaderTrap.untrap(modalEl);
+        unhoist(modalEl);
         keyboardTrap.untrap(modalEl);
```

Comment:
Be careful, the order of screenreader trap and keyboard trap have been flipped here.

Reply:
Doesn't it make sense to undo things in the opposite order of the way they were done? It seems that the traps are working correctly to me. Could you do a quick check on this and see if it's working alright?

Reply:
It would make sense, but in that case wouldn't unhoist come last?

Reply:
good point